src/cond.ts

```diff
@@ -0,0 +1,49 @@
+import { purry } from './purry';
+
+type Pair<T, K> = [(value: T) => boolean, (value: T) => K];
```

Comment:
1. The tuple should be readonly
2. For better documentation you can name each field in the tuple
```suggestion
type Pair<T, K> = readonly [predicate: (value: T) => boolean, mapper: (value: T) => K];
```